[Enhancement] Support consuming from Kafka with IPv6 in routine load task

qingzhongli opened the pull request:
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

Commit:
[Enhancement] Support consuming from Kafka with IPv6 in routine load task

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

Commit:
[Enhancement] Support consuming from Kafka with IPv6 in routine load task

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=38604) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=38604) before we can accept your contribution.<br/><hr/>**liqingzhong** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=38604) it.</sub>

rickif commented on fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:
```diff
@@ -90,42 +90,84 @@ public static void beforeClass() throws Exception {
+                    + ")\n"
+                    + "FROM KAFKA\n"
+                    + "(\n"
+                    + "\"kafka_broker_list\" = \"[kafkahost1]:9092,[kafkahost2]:9092\",\n"
```
Why is the `kafkahost1` wrapped with `[]`?

qingzhongli replied:
It used to indicate that kafka broker is an IPv6 address

rickif replied:
I think the IPV6 address should be enclosed, but the host name shouldn't.

rickif commented on fe/fe-core/src/test/java/com/starrocks/analysis/CreateRoutineLoadStmtTest.java:
```diff
@@ -333,288 +467,656 @@ public void testLoadColumns() {
+        {
+            String createSQL = "CREATE ROUTINE LOAD db0.routine_load_1 ON t1 " +
+                    "PROPERTIES(\"max_error_number\" = \"10\") " +
+                    "FROM KAFKA(\"kafka_broker_list\" = \"xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx\",\"kafka_topic\" = \"topic_0\");";
```
Should the ipv6 address be wrapped with `[]`?

qingzhongli replied:
Thank you for your review, I will fix it

rickif commented on fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateRoutineLoadStmt.java:
```diff
@@ -136,7 +136,7 @@ public class CreateRoutineLoadStmt extends DdlStmt {
 
     private static final String NAME_TYPE = "ROUTINE LOAD NAME";
-    private static final String ENDPOINT_REGEX = "[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]";
+    private static final String ENDPOINT_REGEX = "[-A-Za-z0-9+&@#/%?=~_|!:,.;\\[\\]]+[-A-Za-z0-9+&@#/%=~_|]";
```
How about using a stricter regular expression to match IPv6 address?

qingzhongli replied:
Can you provide an example? I think more strict may lead to more complexity.

kimmking replied:
Variable name "IPV6_REGEX" is better?

qingzhongli replied:
Exactly. I updated the current PR, please review again.

qingzhongli committed:
modify hostname to ipv6 address

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli committed:
modify hostname to ipv6 address

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli committed:
use a separate regular expression to verify IPv6 address of Kafka broker

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli committed:
use a separate regular expression to verify IPv6 address of Kafka broker

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

andyziye committed:
[Tool] Fixed pr was incorrectly merged but SQL-Tester failed (#41109)

Signed-off-by: AndyZiYe <yeziyu@starrocks.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

andyziye committed:
[Tool] Fixed pr was incorrectly merged but SQL-Tester failed (#41109)

Signed-off-by: AndyZiYe <yeziyu@starrocks.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

evelynzhaojie committed:
[Doc]delete unsupported syntax ”with rollup, with cude“ in GROUP BY (#41134)

Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

evelynzhaojie committed:
[Doc]delete unsupported syntax ”with rollup, with cude“ in GROUP BY (#41134)

Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

srlch committed:
[Enhancement] improve perf of dictionary (#40215) (#40216)

Signed-off-by: srlch <linzichao@starrocks.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

srlch committed:
[Enhancement] improve perf of dictionary (#40215) (#40216)

Signed-off-by: srlch <linzichao@starrocks.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

mergify[bot] committed:
[BugFix] in cgroupv2 k8s env,  memory.max is in /sys/fs/cgroup/kubepods (backport #31406) (#41110)

Co-authored-by: dujl <dujlmail@gmail.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

mergify[bot] committed:
[BugFix] in cgroupv2 k8s env,  memory.max is in /sys/fs/cgroup/kubepods (backport #31406) (#41110)

Co-authored-by: dujl <dujlmail@gmail.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

liuyehcf committed:
[Enhancement] Evaluate time functions with constant arguments (#41115)

Signed-off-by: liuyehcf <1559500551@qq.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

liuyehcf committed:
[Enhancement] Evaluate time functions with constant arguments (#41115)

Signed-off-by: liuyehcf <1559500551@qq.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] add keywords to stream load and broker load docs (#41144)


Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] add keywords to stream load and broker load docs (#41144)

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] update stream load doc (#41169)

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] update stream load doc (#41169)


Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

Smith-Cruise committed:
[Enhancement] Add shared buffer align size stat (#41135)

Signed-off-by: Smith Cruise <chendingchao1@126.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

Smith-Cruise committed:
[Enhancement] Add shared buffer align size stat (#41135)

Signed-off-by: Smith Cruise <chendingchao1@126.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

banmoy committed:
[BugFix] Fix wrong read/write stat and metrics (#40407)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

banmoy committed:
[BugFix] Fix wrong read/write stat and metrics (#40407)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

meegoo committed:
[BugFix] Fix spark load load start timestamp not set (#41082)

Signed-off-by: meegoo <meegoo.sr@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

meegoo committed:
[BugFix] Fix spark load load start timestamp not set (#41082)

Signed-off-by: meegoo <meegoo.sr@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

trueeyu committed:
[BugFix] Fixed the mem statistics problem of manual compaction (#41150)

Why I'm doing:
There may be deviation in memory statistics for each compaction task, so when finishing the task, the held MemTracker must be destructed so that errors will not accumulate. However, manul compaction for the primary key model and other model with enable_size_tiered_compaction_strategy=false, is actually using
global compaciton mem tracker, which is a global mem tracker and will not be destructed, which will lead to accumulation of deviation so the statistical compaction memory will continue to increase.

What I'm doing:
Create one child mem tracker for each compaction task
The code of manual compaction is added after fork, so remove the old invalid license.

Signed-off-by: trueeyu <lxhhust350@qq.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

trueeyu committed:
[BugFix] Fixed the mem statistics problem of manual compaction (#41150)

Why I'm doing:
There may be deviation in memory statistics for each compaction task, so when finishing the task, the held MemTracker must be destructed so that errors will not accumulate. However, manul compaction for the primary key model and other model with enable_size_tiered_compaction_strategy=false, is actually using
global compaciton mem tracker, which is a global mem tracker and will not be destructed, which will lead to accumulation of deviation so the statistical compaction memory will continue to increase.

What I'm doing:
Create one child mem tracker for each compaction task
The code of manual compaction is added after fork, so remove the old invalid license.

Signed-off-by: trueeyu <lxhhust350@qq.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

sduzh committed:
[Enhancement] (1/n) Improve data cleanup performance for dropped tables in shared data mode (#39883)

Signed-off-by: Alex Zhu <zhuming9011@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

sduzh committed:
[Enhancement] (1/n) Improve data cleanup performance for dropped tables in shared data mode (#39883)

Signed-off-by: Alex Zhu <zhuming9011@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli commented:
@Mergifyio rebase

qingzhongli commented:
@Mergifyio rebase

TszKitLo40 committed:
[BugFix] Fix stuck when transform from batch to single publish (#40679)

Signed-off-by: Zijie Lu <wslzj40@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

TszKitLo40 committed:
[BugFix] Fix stuck when transform from batch to single publish (#40679)

Signed-off-by: Zijie Lu <wslzj40@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

TszKitLo40 committed:
[Enhancement] Add storage volume in tables_config (#40853)

Signed-off-by: Zijie Lu <wslzj40@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

TszKitLo40 committed:
[Enhancement] Add storage volume in tables_config (#40853)

Signed-off-by: Zijie Lu <wslzj40@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

silverbullet233 committed:
[Enhancement] reduce io times in ColumnarSerde::deserialize (#40800)

Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

silverbullet233 committed:
[Enhancement] reduce io times in ColumnarSerde::deserialize (#40800)

Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

fzhedu committed:
[Feature] concurrently jit and put compiled func into LRU cache (#40355)

Signed-off-by: Zhuhe Fang <fzhedu@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

fzhedu committed:
[Feature] concurrently jit and put compiled func into LRU cache (#40355)

Signed-off-by: Zhuhe Fang <fzhedu@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

fzhedu committed:
[Feature] jit logical and, or, not (#41127)

Signed-off-by: Zhuhe Fang <fzhedu@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

fzhedu committed:
[Feature] jit logical and, or, not (#41127)

Signed-off-by: Zhuhe Fang <fzhedu@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

Seaven committed:
[BugFix] Fix blank json path (#41167)

Signed-off-by: Seaven <seaven_7@qq.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

Seaven committed:
[BugFix] Fix blank json path (#41167)

Signed-off-by: Seaven <seaven_7@qq.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] add note to stream load transaction interface doc (#41207)

Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

amber-create committed:
[Doc] add note to stream load transaction interface doc (#41207)


Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

before-Sunrise committed:
[UT] fix sql test (#41156)

Signed-off-by: before-Sunrise <unclejyj@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

before-Sunrise committed:
[UT] fix sql test (#41156)

Signed-off-by: before-Sunrise <unclejyj@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

meegoo committed:
[BugFix] Fix data ingestion stuck when the new automatic partition and the old partition are distributed on different backends (#40638)

Signed-off-by: meegoo <meegoo.sr@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

meegoo committed:
[BugFix] Fix data ingestion stuck when the new automatic partition and the old partition are distributed on different backends (#40638)

Signed-off-by: meegoo <meegoo.sr@gmail.com>
Signed-off-by: liqingzhong <liqingzhong@inspur.com>
Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

rickif commented:
@qingzhongli 
Please sign the CLA for all commits.
You could run `git rebase HEAD~8 --signoff` and submit again.

mingmxu committed:
[Enhancement] use static link for latest datadog ddprof download link (#41049)

Signed-off-by: Mingmin Xu <m.xu@celonis.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

TszKitLo40 committed:
[UT] Refactor some lake tests by using StarRocks mini cluster (#41202)

Signed-off-by: Zijie Lu <wslzj40@gmail.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli committed:
Merge branch 'StarRocks:main' into kafka_with_ipv6

murphyatwork committed:
[BugFix] fix resultset metadata of prepare protocol (#41080)

Signed-off-by: Murphy <mofei@starrocks.com>Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

qingzhongli committed:
Merge branch 'StarRocks:main' into kafka_with_ipv6

qingzhongli committed:
Merge branch 'kafka_with_ipv6' of https://github.com/qingzhongli/starrocks into kafka_with_ipv6

qingzhongli commented:
@rickif @kimmking Thank you for your patient reviews. This PR has some mistakes, I will close it and submit a new PR.
